src/platform/packages/shared/kbn-sse-utils-server/src/observable_into_event_source_stream.ts

```diff
@@ -54,21 +65,24 @@ export function observableIntoEventSourceStream(
     }),
     map((event) => {
       const { type, ...rest } = event;
-      return `event: ${type}\ndata: ${JSON.stringify(rest)}\n\n`;
+      return createLine({ event: type, data: rest });
     })
   );
 
-  const stream = new PassThrough();
+  const stream = new ResponseStream();
 
   const intervalId = setInterval(() => {
     // `:` denotes a comment - this is to keep the connection open
     // it will be ignored by the SSE parser on the client
-    stream.write(': keep-alive');
+    stream.write(': keep-alive\n');
+    stream.flush();
   }, 10000);
 
   const subscription = withSerializedErrors$.subscribe({
     next: (line) => {
       stream.write(line);
+      // Make sure to flush the written lines to emit them immediately (instead of waiting for buffer to fill)
+      stream.flush();
```

Comment:
@dgieselaar just making sure: do you think it's fine flushing at every event?

Reply:
Thanks, probably not if it impacts performance. I did not realize this. I need to do some research here about how buffering works - any recommended reads will be appreciated but I will try to figure it out myself.

Reply:
if there's any concern, we could apply a debounce to the flushes (in case that we're emitting data too fast, with a max wait of `Xms`, where X is TBD?)

Reply:
Regarding the docs: https://nodejs.org/api/zlib.html#flushing

Reply:
Yeah, I have no idea if it impacts performance, but I thought it might, depending on what exactly is done on flush

Reply:
looks good now, thanks!